### PR TITLE
Externalize protobuf as Maven artifact

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Related issue(s)**:
 - Addresses #883 from Services POV (doc will be updated in separate PR as other repos are finalized).

**Summary of the change**:
 - Remove all _*.proto_, rename _hapi-proto/_ to _hapi-utils/_ (retains a handful of legacy classes, e.g. `FeeBuilder` variants).
 - In _pom.xml_:
   - Add top-level dependency on `com.hedera.hashgraph:hedera-protobuf-java-api:0.12.0-SNAPSHOT`.
   - Add OSSRH snapshot repository under `repositories` element.

**External impacts**:
 - None intended, but note that if you are working on a branch in Services that needs _*.proto_ changes not yet published in the `0.12.0-SNAPSHOT` artifact, you need to publish e.g. a `0.12.0-issue.123-SNAPSHOT` version with your changes and update the _pom.xml_ in your branch accordingly.

### CODE REVIEW

File: hapi-fees/pom.xml [hunk 1]
```diff
@@ -20,7 +20,7 @@
   <dependencies>
     <dependency>
       <groupId>com.hedera.hashgraph</groupId>
-      <artifactId>hedera-protobuf-java-api</artifactId>
+      <artifactId>hapi-utils</artifactId>
       <version>0.11.1-SNAPSHOT</version>
```
Comment (ljianghedera) [inline hapi-fees/pom.xml:24]: Should the version be `0.12.x-SNAPSHOT` or do we want this changes to be in 0.11.x release?

Comment (tinker-michaelj) [reply hapi-fees/pom.xml:24]: The version for the `hapi-utils` artifact will always match the parent version in the Services _pom.xml_ (just like for `hapi-fees`, `test-clients`, etc).

Comment (ljianghedera) [reply hapi-fees/pom.xml:24]: I see. My bad. I thought the `hapi-protobuf` is already in `0.12.0`, so I assume `hedera-services` shall also be `0.12.0`.

**Review by ljianghedera — APPROVED** [review]
LGTM.

**Review by JeffreyDallas — APPROVED** [review]

**Review by anighanta — APPROVED** [review]

**Review by qnswirlds — APPROVED** [review]
LGTM

**Pull request merged**